azure-pipeline/src/index.ts

```diff
@@ -0,0 +1,113 @@
+import taskLib = require('azure-pipelines-task-lib/task');
+import { AxiosRequestConfig  } from "axios";
+import { updateUserConfig } from './utils/api';
+import {
+    API_VERSION,
+    SCOPE,
+    OUTPUT_PATH,
+    INPUT,
+} from './utils/constants';
+import {
+    FileHandler,
+} from './utils/file-handler';
+import {
+    validatedEnvVar,
+    validatedAuthParams,
+} from './utils/common';
+import { 
+    ClientSecretCredential,
+    AuthenticationError,
+} from '@azure/identity';
+import { info } from 'console';
+
+
+class ConfigUpdater {
+
+    private f = new FileHandler;
+
+    // getAuthorizationTokenFromKey():
```

Comment:
👍 for adding docstrings to your methods-- I'd also like to see docstrings else where like on the class to explain what it does but, nit I guess.

I don't think repeating the method signature in the docstring is necessary, but maybe this is a style I am unaware of? This method might return _something_-- also _might_ throw an error-- but what that is is not documented.  I am not a typescript person so maybe this is usual and is already known some how other than reading the code?

Reply:
Great check! I found that the docstrings are a bit messy when it's firstly wrote. I reorganized all docstrings accross the project.